packages/babel-helpers/src/helpers/wrapRegExp.ts

```diff
@@ -72,9 +72,18 @@ export default function _wrapRegExp(this: any): RegExp {
       ).call(
         this,
         str,
-        substitution.replace(/\$<([^>]+)>/g, function (_, name) {
-          var group = groups[name];
-          return "$" + (Array.isArray(group) ? group.join("$") : group);
+        substitution.replace(/\$<([^>]+)(>|$)/g, function (match, name, end) {
+          if (end === "") {
+            // return unterminated group name as-is
+            return match;
+          } else {
+            var group = groups[name];
+            return Array.isArray(group)
+              ? "$" + group.join("$")
+              : typeof group === "number"
```

Comment:
This check relies on the observation that all builtin object prototype fields are non-number, so that we don't have to emit `{ __proto__: null }` to every `wrapRegExp` calls. Of course this check will break if user modified the object prototype, but that is the case that we do not plan to support.